compare-bug-version-and-git-branch.sh

```diff
@@ -9,73 +9,99 @@ DEBUG=${DEBUG:=0}
+    echo ""
+    echo "Submissions that are bug fixes are tracked using Bugzilla [2]."
+    echo ""
+    echo "A check on the commit message, reveals that there is no bug or"
+    echo "github issue referenced in the commit message"
+    echo ""
+    echo "[1] https://github.com/gluster/glusterfs/issues/new"
+    echo "[2] https://bugzilla.redhat.com/enter_bug.cgi?product=GlusterFS"
+    echo ""
+    echo "Please file an github issue or bug and reference the same in the"
+    echo "commit message using the following tags:"
+    echo "For Github issues:"
+    echo "\"Fixes: gluster/glusterfs#n\" OR \"Updates: gluster/glusterfs#n\" OR"
+    echo "\"Fixes: #n\" OR \"Updates: #n\","
+    echo "For a Bug fix:"
+    echo "\"Fixes: bz#n\" OR \"Updates: bz#n\","
+    echo "where 'n' is the issue number or a bug id"
+    echo ""
+    echo "Please resubmit your patch with reference to get +1 vote from this job"
     exit 1
-elif ! grep -q -e '^master$' -e '^release-' <<< "${GERRIT_BRANCH}" ; then
-    echo "Branch '${GERRIT_BRANCH}' can not be mapped to a version, assuming testing only."
-    exit 0
 fi
 
 # Query bugzilla with 3 retries
 [ "${DEBUG}" == '0' ] || BZQOPTS='--verbose'
 BUG_PRODUCT=""
 BZQTRY=0
-while [ -z "${BUG_PRODUCT}" -a ${BZQTRY} -le 3 ]; do
-    BZQTRY=$((${BZQTRY} + 1))
+while [ -z "${BUG_PRODUCT}" ] && [ ${BZQTRY} -le 3 ]; do
+    BZQTRY=$((BZQTRY+1))
     if [ "x${BZQTRY}" = "x3" ]; then
         echo "Failed to get details for BUG id ${BUG}, please verify the bug is not private"
         echo "If the bug is public and readable, please email gluster-infra@gluster.org."
         echo 1
     fi
 
-    BZQOUT=$(bugzilla ${BZQOPTS} query -b ${BUG} --outputformat='%{product}:%{version}:%{groups}:%{status}')
+    BZQOUT=$(bugzilla ${BZQOPTS} query -b "${BUG}" --outputformat='%{product}:%{version}:%{groups}:%{status}')
     BUG_PRODUCT=$(cut -d: -f1 <<< "${BZQOUT}")
     BUG_VERSION=$(cut -d: -f2 <<< "${BZQOUT}")
     BUG_GROUPS=$(cut -d: -f3 <<< "${BZQOUT}")
     BUG_STATUS=$(cut -d: -f4 <<< "${BZQOUT}")
 done
 
-if [ "${BUG_STATUS}" != "NEW" -a "${BUG_STATUS}" != "POST" -a "${BUG_STATUS}" != "ASSIGNED" ]; then
-    echo "BUG id ${BUG} has an invalid status as ${BUG_STATUS}. Acceptable status values are NEW, ASSIGNED or POST."
+if [ "${BUG_PRODUCT}" != "GlusterFS" ]; then
+    echo "BUG id ${BUG} belongs to '${BUG_PRODUCT}' and not 'GlusterFS'."
     exit 1
 fi
 
-if [ "${BUG_GROUPS}" != '[]' ]; then
-    echo "BUG id ${BUG} is marked private, please remove the groups."
+if [ "${BUG_STATUS}" != "NEW" ] && [ "${BUG_STATUS}" != "POST" ] && [ "${BUG_STATUS}" != "ASSIGNED" ] && [ "${BUG_STATUS}" != "MODIFIED" ]; then
+    echo "BUG id ${BUG} has an invalid status as ${BUG_STATUS}. Acceptable status values are NEW, ASSIGNED, POST or MODIFIED"
     exit 1
 fi
 
-if [ "${BUG_PRODUCT}" != "GlusterFS" ]; then
-    echo "BUG id ${BUG} belongs to '${BUG_PRODUCT}' and not 'GlusterFS'."
+if [ "${BUG_GROUPS}" != '[]' ]; then
```

Comment:
This will need to be the first check. Because if the bug is private, I'm pretty sure you can't do any of the following checks.

Reply:
I guess, it should work. Because the status check was before this check even earlier, and it worked fine.